File: Demo/ObjCDemo/Tools.m
//
//  Tools.m
//  ObjCDemo
//
//  Created by liguoliang on 2021/3/3.
//

#import "Tools.h"

@implementation Tools

+ (void)sendMessage {
    NSLog(@"%s", __func__);
}

+ (void)sendMessage:(NSString *)msg from:(NSString *)from to:(NSString *)to {
    NSLog(@"%s, %@->%@ : %@", __func__, from, to, msg);
}

+ (int)sumA:(int)a B:(int)b {
    return a+b;
}
@end

File: Demo/SwiftDemo/Tools.swift
//
//  Tools.swift
//  SwiftDemo
//
//  Created by liguoliang on 2021/3/1.
//

import UIKit

class Tools: NSObject {
    
    class func sendMessage() {
        print("\(#function)")
    }
    
    class func sendMessage(_ message:String) {
        print("\(message)")
    }

    class func sendMessage(_ msg:String, from:String, to:String) -> Void {
        print("\(#function), \(from)->\(to) : \(msg)")
    }

    class func sum(a:Int, b:Int) -> Int {
        a+b;
    }
}
